File: Magik2.0/resource/Models/Stage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Resource.Models
{
    public class Stage
    {
        public int Id { get; set; }

        public int ProjectId { get; set; }

        [StringLength(50)]
        public string Name { get; set; } = null!;

        [StringLength(2048)]
        public string Description { get; set; } = null!;

        public DateTime CreationDate { get; set; }

        public DateTime Deadline { get; set; }

        [Range(0, 100)]
        public int Progress { get; set; }
        

        public Project? Project { get; set; }
    }
}

File: Magik2.0/ui/src/models/resource/stage.ts
import { Attachment } from "./attachment";

export class Stage {
    constructor(public name:string,
                public description:string,
                public deadline:Date,
                public attachments:Attachment[],
                public creationDate?:Date,
                public id?:number,
                public progress?:number,
                public color?:string) {}
}